content/en/blog/_posts/2025-04-21-kubernetes-network-evolution/index.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: blog
+title: "The Evolution of Networking in Kubernetes: Declarative Networking"
+date: 2025-04-21
+draft: false 
+slug: kubernetes-network-evolution
+author: Antonio Ojea (Google)
+---
+
+Kubernetes is in continuous evolution, and as a distributed system, networking is an inseparable aspect, requiring to evolve to support the platform's ongoing development. What began with a relatively simple, flat network model has progressively evolved to accommodate increasingly complex and demanding workloads. Understanding this journey, from the foundational choice of the Container Network Interface (CNI) over Docker's Container Network Model (CNM) to the latest advancements in Dynamic Resource Allocation (DRA) and Declarative Networking, is crucial for anyone operating within the Kubernetes ecosystem. This post illustrates this path, diving into the key decisions and events that have shaped Kubernetes networking, providing insights into the rationale behind the current state and the direction of future development.
+
+## The Genesis: Choosing CNI Over CNM
+
+In the early days of Kubernetes, the question of how to manage container networks was a significant one. Two primary models emerged: the Container Network Interface (CNI) and Docker's Container Network Model (CNM). Ultimately, the Kubernetes community opted for CNI, a decision rooted in fundamental architectural and practical considerations, as explained in Tim Hockin’s blog [Why Kubernetes Doesn't use libnetwork](https://blog.kubernetes.io/2016/01/why-kubernetes-doesnt-use-libnetwork/).
```

Comment:
This summary feels slightly wrong from what I remember. In particular, as noted in the blog post, Kubernetes used neither CNM nor CNI in 1.0. (IIRC kubelet told docker to use its "bridge" plugin when setting up pods, and then it called a shell script to do whatever else was needed, including [possibly removing the pod from the docker bridge](https://github.com/openshift/origin/blob/v1.0.0/Godeps/_workspace/src/github.com/openshift/openshift-sdn/ovssubnet/bin/openshift-ovs-subnet#L40).) I think CNI was inspired by the k8s 1.0 system, and then k8s moved to that. @thockin and @squeed should remember more.

Reply:
I don't see the part that is wrong, is ony explaining that there were mainly two models and one was not chosen